### ISSUE TITLE
Upgrading from 1.1.1 to 1.2 on RHEL/CentOS does not install service

### ISSUE THREAD

**Issue by malarky** [couchbase/sync_gateway#1645]
If I install SG 1.1.1 on Centos 6, and do not use the service install script, then perform an upgrade, the upgrade process does not run the service install script, and therefore doesn't start Sync Gateway automatically, even though it claims that it has:

```
$ sudo rpm -Uvh /vmhost_home/Downloads/couchbase-sync-gateway-enterprise_1.2.0-2399_x86_64.rpm
Preparing...                ########################################### [100%]
   1:couchbase-sync-gateway ########################################### [100%]
sync_gateway start/running, process 8303
sync_gateway stop/waiting

You have successfully uninstalled Couchbase Sync Gateway.


You have successfully installed Couchbase sync_gateway.

You can find sample sync_gateway configuration files in the /opt/couchbase-sync-gateway/examples folder.

You can control the Couchbase sync_gateway service by using the following command:

  initctl start sync_gateway

That's it! sync_gateway is now running on port 4984. We've setup a simple in-memory database
which works great for exploring sync_gateway's capabilities. A limited console is available
by opening your browser to http://localhost:4985/_admin/.

The command-line options are:
    :    :    :
    :    :    :
```

```
$ ps ax | grep sync_gateway
 8404 pts/0    S+     0:00 grep sync_gateway
```

```
$ initctl start sync_gateway
initctl: Unknown job: sync_gateway
```

I have tested the equivalent on Ubuntu 12.04 and it seems to work OK.

**Comment by ajres** [comment]
This is a duplicate of #1160, the service does get installed but the sync_gateway binary gets deleted by the SG 1.1.1. package scripts when 1.1.1 is uninstalled.

There is no practical way to fix 1.1.1, from 1.2 the package scripts are aware of the context of the removal i.e. UPGRADE and will not remove the sync_gateway binary.

**Comment by malarky** [comment]
OK, I hadn't spotted the binary getting removed.  Pretty sure the service isn't there though.  Moot point I guess :-)
